### ISSUE TITLE
the "||" doesn't work well

### ISSUE THREAD

**Issue by kesonIT** [containous/yaegi#465]
"||" doesn't works well

```go
func Test_OR2(t *testing.T) {
	src := `
		package foo
		func Bar(s string) string { 
			var a error = nil
			if a == nil || a.Error() == "nil" {
				println("a is nil")
			}
			return s + "-Foo" 
		}
	`
	i := interp.New(interp.Options{})

	_, err := i.Eval(src)
	if err != nil {
		panic(err)
	}

	v, err := i.Eval("foo.Bar")
	if err != nil {
		panic(err)
	}

	bar := v.Interface().(func(string) string)
	r := bar("Kung")
	println(r)
}
```

Expected result:
```
should print "a is nil"
```

Got:
```
doesn't print 'a is nil'
```

**Comment by kesonIT** [comment]
please take a look at this one, Thanks!

**Comment by mvertes** [comment]
closed in favor of #471
